python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1188,18 +1188,28 @@ def execute_asset_backfill_iteration_inner(
     expensive operations.
     """
     initial_candidates: Set[AssetKeyPartitionKey] = set()
-    request_roots = not asset_backfill_data.requested_runs_for_target_roots
-    if request_roots:
-        initial_candidates.update(
-            asset_backfill_data.get_target_root_asset_partitions(instance_queryer)
-        )
+    # tracks whether this backfill has successfully requested all roots
+    all_roots_requested = asset_backfill_data.requested_runs_for_target_roots
+
+    if not all_roots_requested:
+        backfill_roots = asset_backfill_data.get_target_root_asset_partitions(instance_queryer)
+        # if the previous tick of the backfill did not successfully submit runs (ex code serer unavailable)
+        # we need to re-request the roots
+        not_yet_requested = [
+            root for root in backfill_roots if root not in asset_backfill_data.requested_subset
+        ]
+        if not_yet_requested:
+            initial_candidates.update(not_yet_requested)
+            all_roots_requested = False
+        else:
+            all_roots_requested = True
 
         yield None
 
         updated_materialized_subset = AssetGraphSubset()
         failed_and_downstream_subset = AssetGraphSubset()
         next_latest_storage_id = _get_next_latest_storage_id(instance_queryer)
-    else:
+    if all_roots_requested:
```

Comment:
maybe a comment on looking at `all_roots_requested` instead of an `else` since `all_roots_requested` can change to True in the above condition